File: test/fixtures/accuracy/ruby-project/src/service.rb
require_relative './models'
require_relative './mixins'

class UserService
  def initialize
    @repo = UserRepo.new
  end

  def register(name, email)
    user = User.new(name, email)
    @repo.save(user)
    user
  end
end

class AdminService < UserService
  include Loggable

  def admin_register(name, email)
    log("Registering admin: #{name}")
    user = User.new(name, email)
    @repo.save(user)
    @repo.find(email)
  end
end

File: test/fixtures/accuracy/php-project/src/service.php
<?php

class UserService {
    private UserRepository $repo;

    public function __construct() {
        $this->repo = new UserRepository();
    }

    public function register(string $name, string $email): User {
        $user = new User($name, $email);
        $this->repo->save($user);
        return $user;
    }
}
